src/ast/astBuilder.ts

```diff
@@ -999,33 +998,97 @@ export class ASTBuilder implements CVisitor<any> {
 
   visitPostfixExpression(ctx: PostfixExpressionContext): Expression {
     const primaryExpression = ctx.primaryExpression();
-    const argExpressionList = ctx.argumentExpressionList();
-
-    if (canBeCallExpression(ctx) && primaryExpression !== undefined) {
-      const astPrimaryExpression =
-        this.visitPrimaryExpression(primaryExpression);
-      if (!isIdentifier(astPrimaryExpression)) {
-        throw new BrokenInvariantError(
-          'Encountered an ArgumentExpressionList without an Identifier'
-        );
-      }
-      return {
-        type: 'CallExpression',
-        id: astPrimaryExpression,
-        arguments:
-          argExpressionList.length > 0
-            ? this.visitArgumentExpressionList(argExpressionList[0])
-            : []
-      };
+    if (primaryExpression === undefined) {
+      throw new BrokenInvariantError(
+        'Encountered a PostfixExpression without a PrimaryExpression.'
+      );
     }
 
-    if (primaryExpression !== undefined) {
-      return this.visitPrimaryExpression(primaryExpression);
+    const children = ctx.children;
+    if (children === undefined) {
+      throw new BrokenInvariantError(
+        'Encountered a PostfixExpression with no child nodes.'
+      );
     }
 
-    // TODO: Deal with everything else.
+    let expression = this.visitPrimaryExpression(primaryExpression);
+    // Skip the first child as it is the primary expression.
+    let currChildIdx = 1;
+    let currArrayAccessIndexExpressionIdx = 0;
+    let currArgumentExpressionListIdx = 0;
+    let currIdentifierIdx = 0;
+    while (currChildIdx < children.length) {
+      const operator = children[currChildIdx].toStringTree();
+      switch (operator) {
+        case '[': {
+          const arrayAccessIndexExpression = ctx.expression(
+            currArrayAccessIndexExpressionIdx++
+          );
+          const indexBeingAccessed = this.visitExpression(
+            arrayAccessIndexExpression
+          );
+          expression = {
+            type: 'ArrayAccessExpression',
+            expression,
+            indexBeingAccessed
```

Comment:
Will this always be the index being accessed? What about declarations like `int arr[5];` is 5 the index being accessed? I thought it's the size.